### PR TITLE
add get_available_backends

### PR DESCRIPTION
**Pull request by geohotstan**
I couldn't get it down to a 1-liner...

```
python -c "from tinygrad import Device; print(Device.get_available_backends())"
['METAL', 'GPU', 'CLANG', 'LLVM']
```

since the order is implicit in `["METAL", "AMD", "NV", "CUDA", "QCOM", "GPU", "CLANG", "LLVM"]` I think `device = available_backends[0]` is ok? :s

### CODE REVIEW

**Comment by geohot** [comment]
Opens every device

File: tinygrad/device.py [hunk 1]
```diff
@@ -28,16 +28,18 @@ def __get_canonicalized_item(self, ix:str) -> Compiled:
     return ret
   @property
   def default(self) -> Compiled: return self[self.DEFAULT]
+  def get_available_backends(self) -> List[str]: return list(self._get_available_backends())
+  def _get_available_backends(self) -> Iterator[str]:
```
Comment (geohot) [inline tinygrad/device.py:32]: why? just iterator

Comment (geohotstan) [reply tinygrad/device.py]:
```
python -c "from tinygrad import Device; print(Device.get_available_backends())"
['METAL', 'GPU', 'CLANG', 'LLVM']
```

can't print it
or am I dumb... 😢

**Comment by ryandorrington** [comment]
#6689

**Pull request merged**